contracts/sudoku.scrypt

```diff
@@ -0,0 +1,115 @@
+import "util.scrypt";
+contract Sudoku {
+
+    bytes board;
+
+    static int N = 9;
+    static bytes EMPTY = b'00';
+
+    constructor(bytes board) {
+        this.board = board;
+    }
+
+    function formatCheck(bytes solution) : bool {
+        require(len(solution) == Sudoku.N * Sudoku.N);
+        return true;
+    }
+
+    function merge(bytes solution) : bytes {
+
+        bytes newBoard = this.board;
+        int i = 0;
+        // N - 2
+        loop (9) {
+            int j = 0;
+            loop (9) {
+
+                bytes value = Util.getElemAt(newBoard, this.index(i, j));
+
+                if (value == Sudoku.EMPTY) {
+                    bytes inputValue = Util.getElemAt(solution, this.index(i, j));
+
+                    int elem = Util.fromLEUnsigned(inputValue);
+
+                    require(elem <= 9);
+
+                    newBoard = Util.setElemAt(newBoard, this.index(i, j), inputValue);
+                }
```

Comment:
What's the solution format? Non-zero element in the initial board has to show up as well or not? If yes,
```
 else {
  // check `solution`'s non-empty element is the same with `this.board`, i.e., u cannot change the initial board.
}
```

If not, make sure they are 0.
```
else {
  // check 0
}
```